server/server.py

```diff
@@ -63,39 +65,43 @@
 api_prefix = api_prefix.rstrip('/')
 
 
-# Handle provider-specific environment variables
-if settings.API_PROVIDER == APIProvider.BEDROCK:
-    if not all(
-        [
-            settings.AWS_ACCESS_KEY_ID,
-            settings.AWS_SECRET_ACCESS_KEY,
-            settings.AWS_REGION,
-        ]
-    ):
-        logger.warning('Using Bedrock provider but AWS credentials are missing.')
-    else:
-        # Export AWS credentials to environment if using Bedrock
-        # Ensure these are set in environment for the AnthropicBedrock client
-        os.environ['AWS_ACCESS_KEY_ID'] = settings.AWS_ACCESS_KEY_ID
-        os.environ['AWS_SECRET_ACCESS_KEY'] = settings.AWS_SECRET_ACCESS_KEY
-        os.environ['AWS_REGION'] = settings.AWS_REGION
-        logger.info(
-            f'AWS credentials loaded for Bedrock provider (region: {settings.AWS_REGION})'
-        )
-elif settings.API_PROVIDER == APIProvider.VERTEX:
-    # Get Vertex-specific environment variables
+def setup_provider_environment(tenant_schema: str):
```

Comment:
move provider handling to dedicated file? would also align well with more ai api providers